frontend/src/config/tokens/token.ts

```diff
@@ -0,0 +1,70 @@
+import fs from 'fs';
+import path from 'path';
+
+import type { EthereumAddress, Token } from '@/types/data';
+
+export class TokenMetadata {
+  readonly symbol: string;
+  readonly decimals: number;
+  readonly addresses: TokenAddresses;
+  readonly imageUrl?: string;
+
+  constructor(data: TokenMetadataData) {
+    this.symbol = data.symbol;
+    this.decimals = data.decimals;
+    this.addresses = data.addresses;
+    this.imageUrl = data.imageUrl;
+  }
```

Comment:
This is a good suggestion, and i would prefer this approach if the data being passed to the constructor was not an object that is read and passed directly from a file.
The `TokenMetadataData` in this case is also used for typing the object that is being read from the file as per the latest changes. Since i typed the exported functions from the `config/utils.ts` file, i ended up with a return type of `Record<string, unknown>` for the `readFileJsonContent` function which in the end doesn't match the type `TokenMetadata` but matches the type `TokenMetadataData`.

I tried using some of the TS utilities but was not really successful. 
Do you think this is still necessary?

Reply:
Ah I didn't see that this interface is necessary for typing the json data. Now this seems totally resonable to me.